spark/v3.5/spark/src/test/java/org/apache/iceberg/spark/source/TestSparkScan.java

```diff
@@ -293,6 +293,73 @@ public void testTableWithTwoColStats() throws NoSuchTableException {
     withSQLConf(reportColStatsEnabled, () -> checkColStatisticsReported(scan, 4L, expectedTwoNDVs));
   }
 
+  @TestTemplate
+  public void testMultipleSnapshotsWithColStats() throws NoSuchTableException {
```

Comment:
I think we're missing a test for the case where a statistics file for the snapshot couldn't be found? Let me know if I just missed it.

Reply:
+1 agree with it.

Reply:
The testcase add has it [here](https://github.com/apache/iceberg/pull/11040/files#diff-f8fd314c6b6bd346311bfc9b0001ec6e840c35f2d428920328aad8d4656fb089R347)